Stepper - Seller Experience: Edit email step

markbiek opened the pull request:
#### Proposed Changes

This adds a new step allowing the user to edit their account email address.

Clicking "Edit email" from `wooVerifyEmail` takes you to the `editEmail` step.

Success after clicking "Send a verification to my new email" takes the user back to `wooVerifyEmail`.

#### Testing Instructions

** Success testing **
1. Go to http://calypso.localhost:3000/setup/wooVerifyEmail?siteSlug=[slug]&flags=signup/woo-verify-email and click "Edit email address".
1. Enter a new, unique email address and click "Send a verification to my new email".
1. You should be taken back to the `wooVerifyEmail` step and should receive a verification email.

** Failure testing **
1. Go to http://calypso.localhost:3000/setup/wooVerifyEmail?siteSlug=[slug]&flags=signup/woo-verify-email and click "Edit email address".
1. Enter an email address that's already in use (like bobo@bob.com) and click "Send a verification to my new email".
1. You should see an error notice as pictured below.

Closes #64163

https://user-images.githubusercontent.com/917632/172869660-ccc6bc6b-5124-42cd-afa9-5b3b3afe9adb.mp4

** Initial view **
<img width="1418" alt="image" src="https://user-images.githubusercontent.com/917632/172689319-ab00f85d-c027-4d7d-92a0-5a1ef38603d6.png">

** Error sending verification email **
![CleanShot 2022-06-09 at 10 14 00](https://user-images.githubusercontent.com/917632/172869514-9d091442-01ab-422f-aed9-54a2d3039ee7.jpeg)

markbiek committed:
New step to change your account email address

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7366846

Hi @markbiek, could you please edit the description of this PR and add a screenshot for our translators? Ideally it'd include all of the following strings:
- `Confirm my email later`
- `Enter a new email address, and your current password`
- `New email`
- `Send a verification to my new email`
- `We'll send you an email to the new address to verify that you own it.`
- `Your email change is pending. Please take a moment to check for an email with the subject "[WordPress.com] New Email Address" to confirm your change.`

Thank you in advance!

markbiek commented:
Translation screenshots:
<img width="1418" alt="image" src="https://user-images.githubusercontent.com/917632/172689319-ab00f85d-c027-4d7d-92a0-5a1ef38603d6.png">

<img width="608" alt="image" src="https://user-images.githubusercontent.com/917632/172689564-2b17eb3b-b6de-405d-bf56-8c1224089f82.png">

markbiek committed:
Connect the wooVerifyEmail step to the editEmail step

Click "Edit email" from `wooVerifyEmail` takes you to the `editEmail`
step.

Success after clicking "Send a verification to my new email" takes the
user back to `wooVerifyEmail`.

andres-blanco commented:
I was testing the happy path, I got the verification email but the screen reads `A verification email has been sent to [OLD_EMAIL].` I expected the message to refer to the email I've just entered. (You can see this behaviour in the video you posted).

The email is directed to the correct email tough.

markbiek commented:
> I was testing the happy path, I got the verification email but the screen reads `A verification email has been sent to [OLD_EMAIL].` I expected the message to refer to the email I've just entered. (You can see this behaviour in the video you posted).
> 
> The email is directed to the correct email tough.

@andres-blanco Good catch. I bet the `wooVerifyEmail` step is pulling the current account email.

I probably need to store the edited email somewhere and have `wooVerifyEmail` conditionally load it.

markbiek committed:
Add edit-email property to the onboard store

This is used to store the value entered from the editEmail step

andres-blanco reviewed (approved):
Tested both paths and it's looking good!

markbiek commented:
Thanks @andres-blanco !

I haven't been able to thoroughly test myself because of my working-copy problems but I'll get those WIP commits cleaned up once I've got that figured out.

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7366846).
